modules/endpoint/variables.tf

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "grpc_config" {
+  description = "The path to the full text of the Service Config YAML file."
+  type        = string
+  default     = null
+}
+
+variable "iam_roles" {
+  description = "Authoritative for a given role. Updates the IAM policy to grant a role to a list of members."
+  type        = list(string)
+  default     = []
+}
+
+variable "iam_members" {
+  description = "Authoritative for a given role. Updates the IAM policy to grant a role to a list of members. Other roles within the IAM policy for the instance are preserved."
+  type        = map(list(string))
+  default     = {}
+}
+
+variable "openapi_config" {
```

Comment:
I assume you can only specify a single API definition file, and it's not optional. If it's like that, would it make sense to change to a single variable

```hcl
variable "endpoint_config" {
  type = object({
    spec = string
    path = string
 }) 
```

Where `spec` can be either `openapi` or `grpc`. Or even simpler

```hcl
variable "endpoint_config" {
  type = object({
    grpc = bool
    path = string
 }) 
```

Where default is to use OpenAPI.